src/pages/index.js

```diff
@@ -1,101 +1,245 @@
+/*                                  Popups                                    */
+/* -------------------------------------------------------------------------- */
+const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
   handleProfileEditSubmit
 );
-popupEditForm.setEventListeners();
+editProfilePopup.setEventListeners();
+
+const editAvatarPopup = new PopupWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
+);
+editAvatarPopup.setEventListeners();
 
-const popupAddForm = new PopupWithForm(
+const addCardPopup = new PopupWithForm(
   "#profile-add-modal",
   handleAddCardSubmit
 );
-popupAddForm.setEventListeners();
+addCardPopup.setEventListeners();
 
 const popupImage = new PopupWithImage("#modal__image");
-popupImage.setEventListiners();
+popupImage.setEventListeners();
 
 const userInfo = new UserInfo({
   titleSelector: ".profile__title",
   descriptionSelector: ".profile__description",
+  avatar: ".profile__image",
 });
 
 // FUNCTIONS/////
 function createCard(cardData) {
-  const card = new Card(cardData, "#card-template", handleImageClick);
-  return card.getCardElement();
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleImageClick,
+    handleLikeClicks,
+    handleDeleteSubmit
+  );
+  return card.getView();
 }
 
 function renderCard(cardData) {
   const cardElement = createCard(cardData);
   section.addItem(cardElement);
 }
 
-function handleImageClick(name, link) {
-  popupImage.open(name, link);
+function handleImageClick(cardData) {
+  popupImage.open(cardData);
 }
 
-function handleProfileEditSubmit(data) {
-  userInfo.setUserInfo(data);
-  popupEditForm.close();
+function handleAvatarSubmit(url) {
+  editAvatarPopup.renderLoading(true);
+  api
+    .updateAvatar(url)
+    .then((res) => {
+      userInfo.updateAvatar(res);
+      avatarForm.reset();
+      avatarFormValidator.resetValidation();
+      editAvatarPopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => editAvatarPopup.renderLoading(false));
 }
 
-function handleAddCardSubmit(cardData) {
-  renderCard({ name: cardData.title, link: cardData.url });
-  cardForm.reset();
-  addFormValidator.disableButton();
-  popupEditForm.close();
+function handleProfileEditSubmit({ name, about }) {
+  editProfilePopup.renderLoading(true);
+  api
+    .updateProfileInfo(name, about)
+    .then((userData) => {
+      userInfo.setUserInfo(userData);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
 }
 
-// BUTTONS////
+function handleAddCardSubmit({ title, url }) {
+  addCardPopup.renderLoading(true);
+
+  api
+    .addCard({ name: title, link: url })
+    .then((cardData) => {
+      const cardElement = createCard(cardData);
+      section.addItem(cardElement);
+      addCardPopup.close();
+      cardForm.reset();
+      addFormValidator.resetValidation();
```

Comment:
`resetValidation` is for clearing possible validation errors, when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting. You need to call `disableButton` there.